### PR TITLE
Cleanup on plugin removal

### PR DESCRIPTION
**Pull request by superseacat**
# Description of the PR

+ lifecycle listener is now present
+ the only cleanup done for now is removing the local settings

# Testing
<table>
  <tr>
    <th>unit</th>
    <th>integration</th>
    <th>manual</th>
  </tr>
  <tr>
    <td>
      <ul>
        <li>- [x] new <b>unit</b> tests created</li>
        <li>- [x] all <b>unit</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>integration</b> tests created</li>
        <li>- [ ] all <b>integration</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [x] <b>manual</b> testing went well</li>
      </ul>
    </td>
  </tr>
</table>  
  
# Have you updated the [TESTING.md](/TESTING.md) or other relevant documentation?

- [ ] Yes
- [ ] Not yet. I will do it next.
- [x] Not relevant

### CODE REVIEW

File: src/main/java/fi/aalto/cs/apluscourses/intellij/services/PluginSettings.java [hunk 1]
```diff
@@ -62,28 +72,43 @@ private MainViewModel createNewMainViewModel(@NotNull Project project) {
...
     }
   }
+
+  /**
+   * Method that unsets all the local settings from {@link LocalSettingsNames}.
+   */
+  public void unsetLocalSettings() {
+    Arrays.stream(LocalSettingsNames.values())
+        .map(LocalSettingsNames::getName)
+        .forEach(propertiesManager::unsetValue);
```
Comment (OlliKiljunen) [inline src/main/java/fi/aalto/cs/apluscourses/intellij/services/PluginSettings.java:112]: This is a very neat implementation!

File: src/main/java/fi/aalto/cs/apluscourses/intellij/services/PluginSettings.java [hunk 2]
```diff
@@ -8,16 +8,26 @@
...
+import java.util.Arrays;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 public class PluginSettings implements MainViewModelProvider {
 
-  public interface LocalSettingsNames {
+  public enum LocalSettingsNames {
```
Comment (superseacat) [inline src/main/java/fi/aalto/cs/apluscourses/intellij/services/PluginSettings.java:19]: en enum after all :disappointed:

**Comment by superseacat** [comment]
> Very nice, clean solution, beautiful design, and appropriately tested. Exemplary PR.
> 
:heart: 
> Btw, the name of the PR sounds like our project is nearly done.
> 
:laughing: 
> The issue was labeled with "manual testing", though `TESTING.md` is not edited. I wonder if there should be some changes?

I think it would make the most sense to remove the label, as we decided not to deal with aspects, that touch the settings import, so testing is not really possible (all the handled settings are "hidden")

**Comment by OlliKiljunen** [comment]
> I think it would make the most sense to remove the label, as we decided not to deal with aspects, that touch the settings import, so testing is not really possible (all the handled settings are "hidden")

Ok. Sounds reasonable.

**Pull request merged**